autoextract/aio/client.py

```diff
@@ -175,10 +202,41 @@ async def request_raw(query: Query,
     ``agg_stats`` argument allows to keep track of various stats; pass an
     ``AggStats`` instance, and it'll be updated.
 
+    Additional ``headers`` for the API request can be provided. This headers
+    are included in the request done against the API endpoint: they
+    won't be used in subsequent requests for fetching the URLs provided in
+    the query.
+
+    The default retry policy can be overridden by providing a ``retry_wrapper``
+    function that would be used to wrap the internal request function and
+    that can react to its exceptions by retying the invocation. For example,
+    you could use the following ``retry_3_attempts`` wrapper function to
+    perform three attempts at most::
+
+        def retry_3_attempts(request_fn):
+            async def fn(*args, **kwargs):
+                for _ in range(3):
+                    try:
+                        return await request_fn()
+                    except Exception as e:
+                        exc = e
+                raise exc
+            return fn
```

Comment:
The docs make it much more clear, but I'm still unsure about the API.
So, we're saying in this example:

1. retry_wrapper wraps some unspecified internal request function
2. Signature of this internal function is not documented. 
3. From the example you can figure out that this internal function has no arguments, must be awaited (i.e. it is a coroutine), and returns something which the wrapper should return. Also, in the example wrapper receives *args and **kwargs for some reason.

I think it is a bit under-specified. But at the same time, I'm not sure what's an advatnage of specifying it further - documenting internal function signature, etc. 

Won't saying that we receive tenacity.Retrying object cover the same use cases, require less documentation, and be more clear for users? Or do you see use cases for passing some other request wrappers here?

Reply:
I see your point. Ideally, we should provide even a much simpler retry configuration interface. E.g. a `RetryConfig` object with some simple possibilities, like `number_of_attemps`, `max_time`, `wait_range`. Something very simple to configure and flexible enough. But this was not my goal with this change. My goal here with retrials was been able to provide maximum flexibility for very advances usages (e.g. being able to change logging by intercepting, etc). So I know in advance that this is a bit obscure. I'll change it to `tenacity.AsyncRetrying` as you suggest, it might improve a bit but it will still be obscure and under-documented, not for the "regular" user. I'm going to work on it. 

> Or do you see use cases for passing some other request wrappers here?

In reality, no. 

> Won't saying that we receive tenacity.Retrying object cover the same use cases, require less documentation, and be more clear for users?

Yes, it will probably improve. Let me change it.

Reply:
@kmike done here: https://github.com/scrapinghub/scrapinghub-autoextract/pull/24/commits/136af5acafc821cb862c1e71dd76e7dcf8d65472 . It seems rounder :+1: